From 325512d0de4ff87e835a2a24bff7e044812366df Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 28 Sep 2022 12:31:28 +0300
Subject: [PATCH] [SW-105911] gaudi3: init static configuration in every boot

Some of the variables in hl_fw_boot_ptrs_ctx structure has fixed
values and were assigned based on arc1_persist_dccm_init_done flag.
Since this structure will be persistent or non-persistent based on
boot info skip flag, this patch performs assignment unconditionally.

Change-Id: Idba66b46e7b747aa105bf9cc1d025f88d2a4c076
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 019cb7b..b8e2278 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -449,16 +449,17 @@
 
 	cold_rst_data.data = base_readl(die_cold_rst_flop_2);
 	if (cold_rst_data.arc1_persist_dccm_init_done)
-		return;
+		goto non_zero_init;
 
 	memset((void *)HL_DCCM_BASE, 0, HL_ARC1_DCCM_PERSIST_SECTION_SIZE);
 
+	cold_rst_data.arc1_persist_dccm_init_done = 1;
+	base_writel(cold_rst_data.data, die_cold_rst_flop_2);
+
+non_zero_init:
 	/* Non zero initializers only required for master(die 0) */
 	if (!die_id)
 		arcboot_rom_persist_data_non_zero_init();
-
-	cold_rst_data.arc1_persist_dccm_init_done = 1;
-	base_writel(cold_rst_data.data, die_cold_rst_flop_2);
 }
 
 void arcboot_rom_init(void)
